com.ibm.streamsx.topology/opt/python/packages/streamsx/topology/context.py

```diff
@@ -50,8 +51,33 @@ def submit(ctxtype, graph, config = None, username = None, password = None):
     Returns:
         An output stream of bytes if submitting with JUPYTER, otherwise returns None.
     """    
+    # path to python binary
+    pythonbin = sys.executable
+    pythonreal = os.path.realpath(pythonbin)
+    pythondir = os.path.dirname(pythonbin) 
+    pythonfile = os.path.basename(pythonbin)
+    pythonrealfile = os.path.basename(pythonreal)
+    pythonconfig = pythondir+"/"+pythonfile+"-config"
+    pythonrealconfig = os.path.realpath(pythondir+"/"+pythonrealfile+"-config")
+    pythonversion = python_version()
+
     if config is None:
         config = {}
+
+    # place the fullpaths to the python binary that is running and 
+    # the python-config that will used into the config
+    config["pythonversion"] = {}
+    config["pythonversion"]["version"] = pythonversion
+    config["pythonversion"]["binaries"] = []
+    bf = {}
+    bf["python"] = pythonreal
+    bf["pythonconfig"] = pythonrealconfig
+    config["pythonversion"]["binaries"].append(bf)
+    # also write the python-config path to a file that will be used 
+    # from within pyversion.sh 
+    with open('/tmp/'+os.environ['USER']+'.pythonconfig', 'w') as pf:
```

Comment:
This seems a somewhat unsafe mechanism to pass information between entities. In a shared environment then this will break.